tests/dem/distribution_custom.cc

```diff
@@ -0,0 +1,152 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2019 by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @brief Inserting particles following a custom distribution.
+ */
+
+// Deal.II includes
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+
+#include <deal.II/particles/particle.h>
+
+// Lethe
+#include <dem/dem_solver_parameters.h>
+#include <dem/volume_insertion.h>
+
+// Tests (with common definitions)
+#include <../tests/tests.h>
+
+using namespace dealii;
+
+template <int dim>
+void
+test()
+{
+  // Creating the mesh and refinement
+  parallel::distributed::Triangulation<dim> tr(MPI_COMM_WORLD);
+  int                                       hyper_cube_length = 1;
+  GridGenerator::hyper_cube(tr,
+                            -1 * hyper_cube_length,
+                            hyper_cube_length,
+                            true);
+  int refinement_number = 2;
+  tr.refine_global(refinement_number);
+
+  MappingQ<dim>            mapping(1);
+  DEMSolverParameters<dim> dem_parameters;
+
+  // Defining simulation general parameters
+  dem_parameters.insertion_info.x_min                                = -0.5;
+  dem_parameters.insertion_info.y_min                                = -0.5;
+  dem_parameters.insertion_info.z_min                                = -0.05;
+  dem_parameters.insertion_info.x_max                                = 0.5;
+  dem_parameters.insertion_info.y_max                                = 0.5;
+  dem_parameters.insertion_info.z_max                                = 0.05;
+  dem_parameters.insertion_info.axis_0                               = 0;
+  dem_parameters.insertion_info.axis_1                               = 1;
+  dem_parameters.insertion_info.axis_2                               = 2;
+  dem_parameters.insertion_info.inserted_this_step                   = 1000;
```

Comment:
Or otherwise instead of printing the diameter of every particle I would print the number of particles with diameter 1 and the number with diameter 2

Reply:
If I used 20 particles, there's a chance off generating 20 particles of Diameter=0.0025 (not with the right seed, I know). With 20 particles, we should expect approximatively 2-3 particles with a diameter=0.005 , which is close to zero. 

Maybe a 1000 was a bit overkill. 100 would be better? With 100, we should expect approximatively 12 particles with a diameter=0.0050 .

Reply:
100 would be better